### ISSUE TITLE
Can't gen IDEA file when project root on different disk root.

### ISSUE THREAD

**Issue by sxp** [lihaoyi/mill#905]
https://github.com/lihaoyi/mill/blob/7d1221fb0e21330659584a8358244d674e9f0d3a/scalalib/src/GenIdeaImpl.scala#L502

OS: Win 10
For example:
My project root at D:\project\test, run `mill mill.scalalib.GenIdea/idea` will got exception:

```
mill.scalalib.GenIdea.idea java.lang.IllegalArgumentException: 'other' has different root
java.base/sun.nio.fs.WindowsPath.relativize(WindowsPath.java:404)
java.base/sun.nio.fs.WindowsPath.relativize(WindowsPath.java:42)
os.Path.relativeTo(Path.scala:456)
mill.scalalib.GenIdeaImpl.url$1(GenIdeaImpl.scala:502)
mill.scalalib.GenIdeaImpl.libraryXmlTemplate(GenIdeaImpl.scala:520)
mill.scalalib.GenIdeaImpl.$anonfun$xmlFileLayout$88(GenIdeaImpl.scala:370)
scala.collection.immutable.List.map(List.scala:246)
scala.collection.immutable.List.map(List.scala:79)
mill.scalalib.GenIdeaImpl.$anonfun$xmlFileLayout$87(GenIdeaImpl.scala:368)
scala.collection.StrictOptimizedIterableOps.flatMap(StrictOptimizedIterableOps.scala:117)
scala.collection.StrictOptimizedIterableOps.flatMap$(StrictOptimizedIterableOps.scala:104)
scala.collection.immutable.Vector.flatMap(Vector.scala:113)
mill.scalalib.GenIdeaImpl.xmlFileLayout(GenIdeaImpl.scala:360)
mill.scalalib.GenIdeaImpl.run(GenIdeaImpl.scala:36)
mill.scalalib.GenIdea$.$anonfun$idea$2(GenIdea.scala:14)
scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
```

**Comment by linuxayn** [comment]
I have the same problem. The coursier_cache directory is typically on C:\, which I think causes this problem.

Back to version 0.6.x is fine.

**Comment by lefou** [comment]
I tried to fix this in pull request #936. Would be great if you could try it out.

**Comment by linuxayn** [comment]
> I tried to fix this in pull request #936. Would be great if you could try it out.

@lefou , thank you! It is working fine now.
